ngraph/frontend/paddlepaddle/src/op/reduce_ops.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <node_context.hpp>
+#include "default_opset.hpp"
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+
+template <typename T>
+NamedOutputs reduce_ops(const NodeContext& node) {
+    auto x = node.get_ng_input("X");
+    auto keep_dim = node.get_attribute<bool>("keep_dim");
+    auto reduce_all = node.get_attribute<bool>("reduce_all", false);
+    Shape input_shape = x.get_shape();
```

Comment:
X may have dynamic shape, which means we cannot get its shape here. Please refer to elementwise_ops.cpp get_partial_shape() about how to get the shape rank.